### ISSUE TITLE
[ARM] Build failed

### ISSUE THREAD

**Issue by sergiobenrocha2** [libretro/pcsx_rearmed#65]
The build broke between 1cc8c85 and f82bcc6

```
cc -o pcsx_rearmed_libretro.so libpcsxcore/cdriso.o libpcsxcore/cdrom.o libpcsxcore/cheat.o libpcsxcore/debug.o libpcsxcore/decode_xa.o libpcsxcore/disr3000a.o libpcsxcore/mdec.o libpcsxcore/misc.o libpcsxcore/plugins.o libpcsxcore/ppf.o libpcsxcore/psxbios.o libpcsxcore/psxcommon.o libpcsxcore/psxcounters.o libpcsxcore/psxdma.o libpcsxcore/psxhle.o libpcsxcore/psxhw.o libpcsxcore/psxinterpreter.o libpcsxcore/psxmem.o libpcsxcore/r3000a.o libpcsxcore/sio.o libpcsxcore/socket.o libpcsxcore/spu.o libpcsxcore/gte.o libpcsxcore/gte_nf.o libpcsxcore/gte_divider.o libpcsxcore/gte_arm.o libpcsxcore/gte_neon.o libpcsxcore/new_dynarec/new_dynarec.o libpcsxcore/new_dynarec/linkage_arm.o libpcsxcore/new_dynarec/pcsxmem.o libpcsxcore/new_dynarec/emu_if.o plugins/dfsound/dma.o plugins/dfsound/freeze.o plugins/dfsound/registers.o plugins/dfsound/spu.o plugins/dfsound/out.o plugins/dfsound/nullsnd.o plugins/dfsound/arm_utils.o plugins/gpulib/gpu.o plugins/gpulib/vout_pl.o plugins/gpu_neon/psx_gpu_if.o plugins/gpu_neon/psx_gpu/psx_gpu_arm_neon.o plugins/cdrcimg/cdrcimg.o plugins/dfinput/main.o plugins/dfinput/pad.o plugins/dfinput/guncon.o frontend/cspace.o frontend/cspace_neon.o frontend/libretro.o frontend/main.o frontend/plugin.o -Wl,-Bsymbolic-functions -Wl,-z,relro -shared -Wl,--no-undefined -lpthread  -lz 
plugins/cdrcimg/cdrcimg.o: In function `CDRinit':
/«BUILDDIR»/libretro-pcsx-rearmed-22+r28~ubuntu14.04.1/plugins/cdrcimg/cdrcimg.c:292: undefined reference to `dlopen'
/«BUILDDIR»/libretro-pcsx-rearmed-22+r28~ubuntu14.04.1/plugins/cdrcimg/cdrcimg.c:296: undefined reference to `dlsym'
/«BUILDDIR»/libretro-pcsx-rearmed-22+r28~ubuntu14.04.1/plugins/cdrcimg/cdrcimg.c:298: undefined reference to `dlerror'
/«BUILDDIR»/libretro-pcsx-rearmed-22+r28~ubuntu14.04.1/plugins/cdrcimg/cdrcimg.c:299: undefined reference to `dlclose'
/«BUILDDIR»/libretro-pcsx-rearmed-22+r28~ubuntu14.04.1/plugins/cdrcimg/cdrcimg.c:294: undefined reference to `dlopen'
frontend/main.o: In function `SysLoadLibrary':
/«BUILDDIR»/libretro-pcsx-rearmed-22+r28~ubuntu14.04.1/frontend/main.c:990: undefined reference to `dlopen'
/«BUILDDIR»/libretro-pcsx-rearmed-22+r28~ubuntu14.04.1/frontend/main.c:992: undefined reference to `dlerror'
frontend/main.o: In function `SysLoadSym':
/«BUILDDIR»/libretro-pcsx-rearmed-22+r28~ubuntu14.04.1/frontend/main.c:1007: undefined reference to `dlsym'
frontend/main.o: In function `SysLibError':
/«BUILDDIR»/libretro-pcsx-rearmed-22+r28~ubuntu14.04.1/frontend/main.c:1017: undefined reference to `dlerror'
frontend/main.o: In function `SysCloseLibrary':
/«BUILDDIR»/libretro-pcsx-rearmed-22+r28~ubuntu14.04.1/frontend/main.c:1030: undefined reference to `dlclose'
collect2: error: ld returned 1 exit status
```

https://launchpadlibrarian.net/284011092/buildlog_ubuntu-trusty-armhf.libretro-pcsx-rearmed_22+r28~ubuntu14.04.1_BUILDING.txt.gz

**Comment by bochecha** [comment]
This isn't an ARM issue, build is broken on x86_64 as well for me, same error messages.

It is due to this commit: https://github.com/libretro/pcsx_rearmed/commit/f82bcc681184536aa6d3ee410b938671c1ba2ead

Can this commit be reverted?

**Comment by twinaphex** [comment]
We should get rid of dynamic library loading, it serves no purpose for our core.

I guess for now we can revert it but long-term I want this gone.

**Comment by bochecha** [comment]
@sergiobenrocha2 Was this fixed? If so, how?

Just asking to better understand what to do in GNOME Games. :)

**Comment by sergiobenrocha2** [comment]
It's building now, I'm not sure where it was fixed

https://code.launchpad.net/~libretro/+recipe/pcsx-rearmed-libretro-daily-git

**Comment by bochecha** [comment]
I guess the fix here was 8356146f0b98c811a0e254b3f35c7e7dc653efce which reverted the problematic commit, as @twinaphex had mentioned.

Thanks for doing that!
